pydicom/dataset.py

```diff
@@ -820,19 +820,23 @@ def convert_pixel_data(self):
                     last_exception = e
                     continue
             if not successfully_read_pixel_data:
-                handlers_tried = " ".join(
-                    [str(x) for x in pydicom.config.image_handlers])
-                logger.info("%s did not support this transfer syntax",
-                            handlers_tried)
-                self._pixel_array = None
-                self._pixel_id = None
-                if last_exception:
-                    raise last_exception
-                else:
-                    msg = ("No available image handler could "
-                           "decode this transfer syntax {}".format(
-                               self.file_meta.TransferSyntaxUID.name))
-                    raise NotImplementedError(msg)
+				if self.PixelData.find('PADDING') > 0:
+                    self.PixelData = self.PixelData[0:len(self.PixelData) - 15] + self.PixelData[len(self.PixelData) - 2:len(self.PixelData)]
```

Comment:
This is quite a strange thing - where does this come from? Is this something inherent in the JPEG2000 encoding?
If this is something standard, please add a respective comment - otherwise we have to understand it first.

Reply:
The dicoms come from a very famous PACS system vender in China called Winning Health Technology(http://www.winning.com.cn). Most hospital in Shanghai use their PACS system. Currently we used pydicom to work on these dicom files and found the issue.


About the Encoding standard, I found out some thing related, but I am not sure it's the correct one. The source link is http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_A.4.html.
![_20180711094811](https://user-images.githubusercontent.com/18344017/42546329-8c9cad5c-84ef-11e8-8ac0-b27356ab25fe.png)

Reply:
The file you provided seems to have been encoded with the kakadu JPEG 2000 library.  My copy of gdcm also fails to decode this.  At the end of the JPEG 2000 stream, there is a section with:

FF 64 00 0B 00 01 50 41 44 44 49 4E 47 FF D9

50 to 47 is the word "PADDING"  

FF 64 is a valid JPEG 2000 marker for a comment .  I think FF D9 is a marker for the end of the comment (even though 00 0B encodes the length?)

This should be a perfectly fine JP2K image, but for some reason, openJPEG or GDCM (I think GDCM uses openJPEG) is not reading it correctly.  

In any case, I don't think this is a case that pydicom should handle in the general release at this  point. 

If we want to pursue this, I would be curious to see if any comment in the JP2K stream causes a failure.  If so, then either we can bump this over to openJPEG to get their opinion, or if it fails, try to decode a second time after removing all JP2K comment tags...  Maybe it is a known issue over there?  I think Pillow might only support the older openJPEG  2.1 (they are up to 2.3 now).